scraping_utils.py

```diff
@@ -293,4 +293,43 @@ def extract_timetables(string, year, term):
         out["courses"].append(course_out)
     return out
 
+def distribute_shared_sections(out):
+    messages = []
+    # For each course
+    for index, course in enumerate(out["courses"]):
+        course_code = course["subject_code"] + course["course_number"]
+        sec_comps = {}
+        for section in course["sections"]:
+            sec_comps[section["id"]] = []
+            for component in section["components"]:
+                if component["type"] not in sec_comps[section["id"]]:
+                    sec_comps[section["id"]].append(component["type"])
 
+        arr_lookup = lambda elt: len(sec_comps[elt["id"]])
+        comp_secs = group_by_eq(course["sections"], arr_lookup)
+
+        if len(comp_secs) >= 1:
```

Comment:
So for most cases, this should do the job, but is there a reason you chose to use `len(comp_secs) >= 1` as opposed to `1 in comp_secs`?  I would have thought that latter would be safer in ensuring that there exists at least one section consisting only of one component?  

This is because the output of `group_by_eq` is a dictionary, which might not have the key `1` in it.  Also, even if it were a list, a length 1 list doesn't have 1 as an index (you'd have to test strictly greater than 1 in that case), and because we use that later on (e.g. line 319).
```suggestion
        if 1 in comp_secs:
```

Reply:
I honestly thought I tried it that way and it didn't work but just tested it and it seems to work so I probably just changed this by mistake while fixing other stuff.